libutils/string_lib.c

```diff
@@ -77,7 +77,7 @@ unsigned int StringHash(const char *str, unsigned int seed, unsigned int max)
     h ^= (h >> 11);
     h += (h << 15);
 
-    return (h & (max - 1));
+    return (h % max);
```

Comment:
Then this means that it is always executed with power-of-two max, so `&` should be working perfectly.